Sources/NIOCore/Channel.swift

```diff
@@ -373,6 +373,16 @@ public enum ChannelError: Error {
     case unremovableHandler
 }
 
+extension ChannelError {
+    // 'any Error' is unconditionally boxed, avoid allocating per use by statically boxing them.
+    static let _alreadyClosed: any Error = ChannelError.alreadyClosed
```

Comment:
I am wondering if should file a compiler bug here and this is something that the compiler ought to be able to optimise. The error enum is in the same module and it has zero associated values. So it might as well store the instance in a global memory region

Reply:
I don't think it's about the underlying error type; holding a `ChannelError` doesn't allocate. It's holding an `any Error` which always allocates.